google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -289,32 +372,48 @@ private PooledSession(Session delegate) {
       markUsed();
     }
 
+    @VisibleForTesting
+    void setAllowDelegateRecreation(boolean allow) {
+      allowDelegateRecreation = allow;
+    }
+
+    private void maybeRecreateUnderlyingSession(SessionNotFoundException e) {
+      if (beforeFirstServerCall && allowDelegateRecreation) {
+        delegate = spanner.createSession(db);
```

Comment:
could we get another session from the pool instead of creating a new one ?

I am afraid that the latency on this request will be too high when we internally retry.